chain/types/ethtypes/eth_types.go

```diff
@@ -815,3 +815,82 @@ func (e EthFeeHistoryParams) MarshalJSON() ([]byte, error) {
 	}
 	return json.Marshal([]interface{}{e.BlkCount, e.NewestBlkNum})
 }
+
+type EthBlockParamByNumberOrHash struct {
+	PredefinedBlock *string
```

Comment:
I'd consider using negative constants to represent "latest", etc. Storing them in the `BlockNumber` field. I.e., we'd have three fields only:

```go
const (
    EthBlockLatest EthUint64 = -iota
    EthBlockPending
    EthBlockEarliest
    // ...
)

type EthBlockParam struct {
    BlockNumber      *EthUint64
    BlockHash        *EthHash
    RequireCanonical bool
}
```

Reply:
Would love feedback from @filecoin-project/lotus-maintainers team on Stebs comment here.